src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    std::string triangles =
+        Triangles::countCentralStoreTriangles(aggregatedCentralStore, distributionHashMap, threadPriority);
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+    return triangles;
+}
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+string JasmineGraphInstanceService::aggregateCompositeCentralStoreTriangles(std::string compositeFileList,
+                                                                            std::string availableFileList,
+                                                                            int threadPriority) {
+    instance_logger.log("###INSTANCE### Started Aggregating Composite Central Store Triangles", "info");
+    std::string aggregatorFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.aggregatefolder");
+    std::string dataFolder = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    map<long, unordered_set<long>> aggregatedCompositeCentralStore;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+    std::vector<std::string> compositeCentralStoreFileList = Utils::split(compositeFileList, ':');
+    std::vector<std::string>::iterator compositeCentralStoreFileIterator;
+    std::vector<std::string> availableCompositeFileList = Utils::split(availableFileList, ':');
+    std::vector<std::string>::iterator availableCompositeFileIterator;
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string workerList = (data);
-            workerList = Utils::trim_copy(workerList, " \f\n\r\t\v");
-            instance_logger.log("Received Worker List " + workerList, "info");
-
-            std::vector<string> workerSockets;
-            stringstream wl(workerList);
-            string intermediate;
-            while (getline(wl, intermediate, ',')) {
-                workerSockets.push_back(intermediate);
-            }
+    for (availableCompositeFileIterator = availableCompositeFileList.begin();
+         availableCompositeFileIterator != availableCompositeFileList.end(); ++availableCompositeFileIterator) {
+        std::string availableCompositeFileName = *availableCompositeFileIterator;
+        size_t lastindex = availableCompositeFileName.find_last_of(".");
+        string rawFileName = availableCompositeFileName.substr(0, lastindex);
+        struct stat st;
```

Comment:
Better if we can find better variable name than `st`

Reply:
If this is used to check if the file exists, we can do it with `access(2)` function with permission checks also.

Reply:
Fixed in the next PR